### PR TITLE
ocamlPackages.camomile: add v2.0.0

### PR DESCRIPTION
**Pull request by Niols**
###### Description of changes

The OCaml package Camomile used to be at [yoriyuki/camomile](https://github.com/yoriyuki/camomile). There is now a fork [ocaml-community/camomile](https://github.com/ocaml-community/camomile) that provides a v2.0.0. This is the one included in OPAM and therefore I think this is the one we should follow in nixpkgs.

Because the API changed so much between Camomile v1 and v2, and because some other packages depend on Camomile, I decided to rename `ocamlPackages.camomile` into `ocamlPackages.camomile_1` (following the change in other packages) and to add a new `ocamlPackages.camomile` package containing v2. I am not sure if that is the right way to do things.

- Release: https://github.com/ocaml-community/Camomile/releases/tag/v2.0.0
- Change log: https://github.com/ocaml-community/Camomile/blob/v2.0.0/CHANGES.md
- Changes: https://github.com/ocaml-community/camomile/compare/d7d8843c88fae774f513610f8e09a613778e64b3...v2.0.0

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/camomile/1.2.0.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ lib, fetchFromGitHub, buildDunePackage, ocaml, cppo }:
+
+buildDunePackage rec {
+  pname = "camomile";
+  version = "1.0.2";
```
Comment (ulrikstrid) [inline pkgs/development/ocaml-modules/camomile/1.2.0.nix:5]: Is the filename or version wrong?

File: pkgs/development/ocaml-modules/camomile/1.2.0.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  useDune2 = true;
```
Comment (ulrikstrid) [inline pkgs/development/ocaml-modules/camomile/1.2.0.nix:7]: Remove or change to `duneVersion = X`  where `X` is the actual version needed.

**Review by ulrikstrid — CHANGES_REQUESTED** [review]
The overall approach seems good :+1:

File: pkgs/development/ocaml-modules/camomile/1.2.0.nix (hunk 1 above)
Comment (wegank) [reply pkgs/development/ocaml-modules/camomile/1.2.0.nix]: The filename is wrong.

**Comment by toastal** [comment]
v2 commit *should* be able to remove the Soupault patch for camomile_v1
-- 
toastal ไข่ดาว | https://toast.al
PGP: 7944 74b7 d236 dab9 c9ef  e7f9 5cce 6f14 66d4 7c9e

File: pkgs/tools/typesetting/soupault/default.nix [hunk 3]
```diff
@@ -24,17 +25,9 @@ ocamlPackages.buildDunePackage {
     sha256 = "nwXyOwDUbkMnyHPrvCvmToyONdbg5kJm2mt5rWrB6HA=";
   };
 
-  patches = lib.lists.optional
```
Comment (toastal) [inline pkgs/tools/typesetting/soupault/default.nix:27]: approve 👍‍

File: pkgs/tools/typesetting/soupault/default.nix [hunk 4]
```diff
@@ -1,6 +1,7 @@
 { lib
+, stdenv
 , fetchFromGitea
-, fetchpatch
+, darwin
```
Comment (Niols) [inline pkgs/tools/typesetting/soupault/default.nix:4]: Why these two additions?

**Review by Niols — COMMENTED** [review]
Thanks for the polishing guys, it looks great! :-)

File: pkgs/tools/typesetting/soupault/default.nix (hunk 4 above)
Comment (wegank) [reply pkgs/tools/typesetting/soupault/default.nix:4]: For fixing the package on darwin. See https://github.com/NixOS/nixpkgs/pull/242103/commits/07f3398415d0737ab441815272735364568a9bf9.

Comment (Niols) [reply pkgs/tools/typesetting/soupault/default.nix:4]: Those I saw, but they are in the file `pkgs/development/ocaml-modules/camomile/default.nix`. My question is about the file `pkgs/tools/typesetting/soupault/default.nix` where these two lines are added but don't seem to be used anywhere.

Comment (wegank) [reply pkgs/tools/typesetting/soupault/default.nix:4]: Oh, my! I'll remove them.

File: pkgs/development/ocaml-modules/camomile/1.0.2.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{ lib, fetchFromGitHub, buildDunePackage, ocaml, cppo }:
+
+buildDunePackage rec {
+  pname = "camomile";
+  version = "1.0.2";
+
+  duneVersion = "2";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/camomile/1.0.2.nix:7]: This is a regression.

Comment (Niols) [reply pkgs/development/ocaml-modules/camomile/1.0.2.nix:7]: How so?

Comment (vbgl) [reply pkgs/development/ocaml-modules/camomile/1.0.2.nix:7]: It used to build fine with the default version of dune.

**Comment by Niols** [comment]
Superseeded by https://github.com/NixOS/nixpkgs/pull/257196. I think this PR should be closed, but some changes involved in it might be to be re-implemented somewhere else.

**Pull request closed without merging**